packages/eslint-plugin/docs/rules/no-wrapper-object-types.mdx

```diff
@@ -0,0 +1,74 @@
+---
+description: 'Disallow using confusing built-in primitive class wrappers.'
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-wrapper-object-types** for documentation.
+
+The JavaScript language includes a set of primitives such as `boolean` and `number`s.
+JavaScript also contains a "wrapper" class object for each of those primitives, such as [`Boolean`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean) and [`Number`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number).
+TypeScript represents the primitive in lower-case types and instances of the wrapper classes with upper-case.
+
+Using the primitives like `0` instead of class constructors like `Number(0)` is generally considered a JavaScript best practice.
```

Comment:
```suggestion
Using the primitives like `0` instead of class constructors like `new Number(0)` is generally considered a JavaScript best practice.
```

`Number()` is the correct way to do coercion.

Reply:
for reference:
```ts
typeof Number('1') === 'number'
typeof new Number('1') === 'object'
```